### PR TITLE
Token handling

### PR DESCRIPTION
**Pull request by tsibley**
Please see the commit messages for details.

### CODE REVIEW

File: pycognito/__init__.py [hunk 1]
```diff
@@ -196,6 +198,12 @@ def __init__(
         else:
             self.client = boto3.client("cognito-idp", **boto3_client_kwargs)
 
+    @property
+    def user_pool_url(self):
+        return "https://cognito-idp.{}.amazonaws.com/{}".format(
+            self.user_pool_region, self.user_pool_id
+        )
```
Comment (pvizeli) [inline pycognito/__init__.py:205]:
```suggestion
        return f"https://cognito-idp.{self.user_pool_region}.amazonaws.com/{self.user_pool_id}")
```

File: pycognito/__init__.py [hunk 2]
```diff
@@ -207,9 +215,7 @@ def get_keys(self):
         # If it is not there use the requests library to get it
         else:
             self.pool_jwk = requests.get(
-                "https://cognito-idp.{}.amazonaws.com/{}/.well-known/jwks.json".format(
-                    self.user_pool_region, self.user_pool_id
-                )
+                self.user_pool_url + "/.well-known/jwks.json"
```
Comment (pvizeli) [inline pycognito/__init__.py:218]:
```suggestion
                f"{self.user_pool_url}/.well-known/jwks.json"
```

File: pycognito/__init__.py [hunk 3]
```diff
@@ -218,26 +224,49 @@ def get_key(self, kid):
...
+                options={
+                    "require_aud": True,
+                    "require_iss": True,
+                    "require_exp": True,
+                },
             )
         except JWTError:
             raise TokenVerificationException(
-                "Your {} token could not be verified."
+                "Your {!r} token could not be verified.".format(id_name)
```
Comment (pvizeli) [inline pycognito/__init__.py:257]: f-string

File: pycognito/__init__.py [hunk 4]
```diff
@@ -218,26 +224,49 @@ def get_key(self, kid):
...
             ) from None
+
+        token_use_verified = verified.get("token_use") == token_use
+        if not token_use_verified:
+            raise TokenVerificationException(
+                "Your {!r} token use ({!r}) could not be verified.".format(
```
Comment (pvizeli) [inline pycognito/__init__.py:263]: f-string

File: pycognito/__init__.py [hunk 5]
```diff
@@ -218,26 +224,49 @@ def get_key(self, kid):
...
+                    id_name, token_use
+                )
+            )
+
         setattr(self, id_name, token)
+        setattr(self, token_use + "_claims", verified)
```
Comment (pvizeli) [inline pycognito/__init__.py:269]:
```suggestion
        setattr(self, f"{token_use}_claims", verified)
```

**Review by pvizeli — COMMENTED** [review]
Looks good, just a small style change, after that we can merge it 👍

**Review by pvizeli — APPROVED** [review]
Thanks!

**Comment by tsibley** [comment]
Thanks! Repushed with change to f-strings.

**Pull request merged**